ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -453,6 +463,56 @@ private void cleanupOldState() {
     localCommands.ifPresent(lc -> lc.processLocalCommandFiles(serviceContext));
   }
 
+  public void cleanupOldStateDirectories(final KsqlConfig configWithApplicationServer) {
+    final String stateDir = configWithApplicationServer.getKsqlStreamConfigProps().getOrDefault(
+            StreamsConfig.STATE_DIR_CONFIG,
+            StreamsConfig.configDef().defaultValues().get(StreamsConfig.STATE_DIR_CONFIG)).toString();
```

Comment:
This default is a little confusing to me... We confirm that the state directory is available [here](https://github.com/confluentinc/ksql/blob/master/ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlServerMain.java#L112), in ksql server main, but we don't set the config if we end up using the default. This means that if we're using a state store from the default, we can't use the config to access the store name anywhere... 

I assume if we don't set the config there (if it's not already set ofc), it's so users can set the config later ..? But it does make it a little tricky to remember to use the default instead of the config. cc @swist

Reply:
Hmm.. I agree this is indeed a bit weird, maybe overlooked rather than intentional?